src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -3632,6 +3632,65 @@ public sealed class C
             actual.Should().Be(Expected);
         }
 
+        [Fact(DisplayName = "DataModelGenerator accepts a limited oneOf with \"type\": \"null\"")]
+        public void AcceptsLimitedOneOfWithTypeNull()
+        {
+            const string Schema =
+@"{
+  ""type"": ""object"",
+  ""properties"": {
+    ""arrayOrNullProperty"": {
+      ""description"": ""A property that can either be an array or null."",
+      ""oneOf"": [
+        {
+          ""type"": ""array""
+        },
+        {
+          ""type"": ""null""
+        }
+      ],
+      ""items"": {
+        ""type"": ""integer""
+      }
+    }
+  }
+}";
+            const string ExpectedClass =
+@"using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace N
+{
+    [DataContract]
+    [GeneratedCode(""Microsoft.Json.Schema.ToDotNet"", """ + VersionConstants.FileVersion + @""")]
+    public partial class C
+    {
+        /// <summary>
+        /// A property that can either be an array or null.
+        /// </summary>
+        [DataMember(Name = ""arrayOrNullProperty"", IsRequired = false, EmitDefaultValue = false)]
```

Comment:
>false [](start = 91, length = 5)

this is wrong! EmitDefaultValue s/be true in this case! or is that too much work to try to get into this change? :)

Reply:
I see. you always emit EmitDefaultValue = false for items that aren't required. a 'oneOf' type is implicitly required. If you elide the default value of oneType, you lose your understanding of which type it was of the various options. do we get out of the woods here by the simple expedient of marking all these things as required? 

---
In reply to: [245204160](https://github.com/Microsoft/jschema/pull/80#discussion_r245204160) [](ancestors = 245204160)

Reply:
Approving but still interested in the conversation of course.

---
In reply to: [245205247](https://github.com/Microsoft/jschema/pull/80#discussion_r245205247) [](ancestors = 245205247,245204160)

Reply:
I think this is ok as it is. I don't agree that a `oneOf` type is implicitly required. I don't see a problem with saying "This thing can either be an array or it can be null, but if it's missing, treat it as null". And if you do specify a default value (null, or an empty array), and you serialize an object but refrain from serializing the default value, you have not lost your understanding of which type it was. When you deserialize later, the deserializer will say "Hmm, this thing can either be an array or null. Darn, it doesn't appear in the document I'm deserializing. Let's see if there's a default... great! the default is null."

---
In reply to: [245205328](https://github.com/Microsoft/jschema/pull/80#discussion_r245205328) [](ancestors = 245205328,245205247,245204160)

Reply:
But by all means if this is still a concern, hit me up offline.

---
In reply to: [245354556](https://github.com/Microsoft/jschema/pull/80#discussion_r245354556) [](ancestors = 245354556,245205328,245205247,245204160)